etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamSenderHelperTest.groovy

```diff
@@ -0,0 +1,131 @@
+package gov.hhs.cdc.trustedintermediary.external.reportstream
+
+import gov.hhs.cdc.trustedintermediary.context.TestApplicationContext
+import gov.hhs.cdc.trustedintermediary.etor.RSEndpointClient
+import gov.hhs.cdc.trustedintermediary.etor.messages.UnableToSendMessageException
+import gov.hhs.cdc.trustedintermediary.etor.metadata.EtorMetadataStep
+import gov.hhs.cdc.trustedintermediary.external.jackson.Jackson
+import gov.hhs.cdc.trustedintermediary.external.localfile.MockRSEndpointClient
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger
+import gov.hhs.cdc.trustedintermediary.wrappers.MetricMetadata
+import gov.hhs.cdc.trustedintermediary.wrappers.formatter.Formatter
+import gov.hhs.cdc.trustedintermediary.wrappers.formatter.FormatterProcessingException
+import gov.hhs.cdc.trustedintermediary.wrappers.formatter.TypeReference
+import spock.lang.Specification
+
+class ReportStreamSenderHelperTest extends Specification {
+
+    def setup() {
+        TestApplicationContext.reset()
+        TestApplicationContext.init()
+        TestApplicationContext.register(ReportStreamSenderHelper, ReportStreamSenderHelper.getInstance())
+        TestApplicationContext.register(RSEndpointClient, MockRSEndpointClient.getInstance())
+        TestApplicationContext.register(MetricMetadata, Mock(MetricMetadata))
+    }
+
+    def "sendToReportStream works"() {
+        given:
+        def mockFormatter = Mock(Formatter)
+        mockFormatter.convertJsonToObject(_ as String, _ as TypeReference) >> Map.of('submissionId', 'fake-id')
+        TestApplicationContext.register(Formatter, mockFormatter)
+
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        ReportStreamSenderHelper.getInstance().sendToReportStream(_ as String, _ as String, _ as String)
+
+        then:
+        noExceptionThrown()
+        1 * ReportStreamSenderHelper.getInstance().metadata.put(_, EtorMetadataStep.SENT_TO_REPORT_STREAM)
```

Comment:
Shouldn't we assert more than just that we registered an internal metadata step?  It looks like our old test didn't assert anything other than no assertion was thrown, but yeah, I think we should at least assert that our RS client was called once.

E.g. something like
`1 * mockRsClient.requestWatersEndpoint(_ as String, _ as String)`.